go/dlpack.go

```diff
@@ -0,0 +1,464 @@
+package cuvs
+
+// #include <stdio.h>
```

Comment:
This would be easier to follow if we mirrored the existing cuVS directory structures for C/C++: `cuvs/core/dlpack.go`, etc...